presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveStorageFormats.java

```diff
@@ -430,6 +434,83 @@ private void runTimestampQueries(String tableName, List<TimestampAndPrecision> d
         onPresto().executeQuery("DROP TABLE " + tableName);
     }
 
+    @Test(dataProvider = "storageFormatsWithNanosecondPrecision", groups = STORAGE_FORMATS)
+    public void testStructTimestamps(StorageFormat format)
+            throws SQLException
+    {
+        setAdminRole(onPresto().getConnection());
+        ensureDummyExists();
+
+        String tableName = format("test_struct_timestamp_precision_%s_%s", format.getName().toLowerCase(Locale.ENGLISH), randomTableSuffix());
+
+        onPresto().executeQuery(format(
+                "CREATE TABLE %s (\n"
+                        + "id INTEGER\n"
+                        + ",arr ARRAY(TIMESTAMP)\n"
+                        + ",map MAP(TIMESTAMP, TIMESTAMP)\n"
+                        + ",row ROW(col TIMESTAMP)\n"
+                        + ",nested ARRAY(MAP(TIMESTAMP, ROW(col ARRAY(TIMESTAMP))))\n"
+                        + ") WITH (%s)",
+                tableName,
+                format.getStoragePropertiesAsSql()));
+
+        // Insert in a loop because inserting with UNION ALL sometimes makes values invisible to Presto
+        for (TimestampAndPrecision entry : TIMESTAMPS_FROM_HIVE) {
+            onHive().executeQuery(format(
+                    "INSERT INTO %1$s\n"
+                            // insert with SELECT because hive bans array/map/struct functions in VALUES
+                            + "SELECT\n"
+                            + "%3$s\n"
+                            + ",array(%2$s)\n"
```

Comment:
i like the convention of `,` at the line start because it allows be to add new line without modifying the previous line, resulting in cleaner diff

this is not a convention in the project though, so i'd recommend putting the comma at line's end 

also, `\n` is redundant, and reads worse than just a space

Reply:
I like including `\n` because IntelliJ highlights it, so it makes it more obvious when I'm missing whitespace; it's annoying when a test fails because I accidentally wrote `"SELECTid,valueFROM sometable`, and the failure messages aren't usually super helpful. (IntelliJ also includes the newlines if you use "edit SQL fragment" and use multiple lines.)

But as with the commas, it doesn't match the style elsewhere, so I'll just indent everything with spaces instead.